signer/src/stacks_api.rs

```diff
@@ -0,0 +1,265 @@
+    pub node_endpoint: url::Url,
+    /// The client used to make the request.
+    pub client: reqwest::Client,
+}
+
+impl StacksClient {
+    /// Create a new instance of the Stacks client using the entries in the
+    /// signer's `config.toml` and environment variables.
+    ///
+    /// See [`StacksSettings::new_from_config`] for more on overriding
+    /// the entries in `config.toml` using environment variables.
+    pub fn new_from_config() -> Result<Self, Error> {
+        let settings = StacksSettings::new_from_config()?;
+
+        Ok(Self {
+            api_endpoint: settings.api.endpoint,
+            node_endpoint: settings.node.endpoint,
+            client: reqwest::Client::new(),
+        })
+    }
+
+    /// Get Stacks block IDs given the bitcoin block hash. Uses the Stacks API
+    /// via the GET /extended/v2/burn-blocks/:height_or_hash endpoint.
+    ///
+    /// See https://docs.hiro.so/api/get-burn-block
+    async fn get_block_ids(&self, hash: &bitcoin::BlockHash) -> Result<Vec<StacksBlockId>, Error> {
+        // The Stacks API expects the hash to be hex encoded with the
+        // leading 0x in the string, which is not produced by the Display
+        // implementation of bitcoin::hashes::sha256d::Hash (but is for
+        // the Debug implementation).
+        let hash: &bitcoin::hashes::sha256d::Hash = hash.as_raw_hash();
+        let path = format!("/extended/v2/burn-blocks/0x{}", hash);
+        let url = self.api_endpoint.join(&path).map_err(Error::PathParse)?;
+
+        let response = self.client.get(url).timeout(REQUEST_TIMEOUT).send().await?;
+        let resp: GetBurnBlockResponse = response.json().await?;
+
+        // The Stacks API often returns hex prefixed with 0x. If this is
+        // the case, we split it off before constructing the block ids.
+        resp.stacks_blocks
+            .into_iter()
+            .map(|hex_string| {
+                let hex_str: &str = if hex_string.starts_with("0x") {
+                    hex_string.split_at(2).1
+                } else {
+                    &hex_string
+                };
+
+                StacksBlockId::from_hex(hex_str)
+                    .map_err(|err| Error::ParseStacksBlockId(err, hex_string))
+            })
+            .collect()
+    }
+
+    /// Fetch the raw stacks nakamoto block from a Stacks node given the
+    /// Stacks block ID.
+    async fn get_block(&self, block_id: &StacksBlockId) -> Result<NakamotoBlock, Error> {
+        let path = format!("/v3/blocks/{}", block_id.to_hex());
+        let url = self.node_endpoint.join(&path).map_err(Error::PathParse)?;
+
+        // TODO: Add more context to these errors
+        let response = self.client.get(url).timeout(REQUEST_TIMEOUT).send().await?;
+        let resp = response.bytes().await?;
+
+        NakamotoBlock::consensus_deserialize(&mut &*resp)
+            .map_err(|err| Error::DecodeNakamotoBlock(err, *block_id))
+    }
+}
+
+impl StacksInteract for StacksClient {
```

Comment:
I think it would be nice to move the `StacksInteract` trait definition to this module now when the module exists. It was only created in the `block_observer` as a placeholder for the "real" trait.

Also, as per #44 we will need the ability to broadcast transactions and estimate fee rates in this trait. Is that something that you think would be easy to add now or do you prefer to create follow-up tickets for it?

Reply:
> I think it would be nice to move the `StacksInteract` trait definition to this module now when the module exists. It was only created in the `block_observer` as a placeholder for the "real" trait.

Yeah, I'll move it here.
 
> Also, as per #44 we will need the ability to broadcast transactions and estimate fee rates in this trait. Is that something that you think would be easy to add now or do you prefer to create follow-up tickets for it?

I think that work should be done in a follow-up PR addressing another ticket. I had thought that the ticket this PR addresses was only about updating Stacks blocks. I'll  open new ones for broadcasting stacks transactions and estimating fees. And if #176 needs to be modified to reflect what this PR addresses I'll do that too.

Reply:
> I think that work should be done in a follow-up PR addressing another ticket. I had thought that the ticket this PR addresses was only about updating Stacks blocks. I'll open new ones for broadcasting stacks transactions and estimating fees. And if https://github.com/stacks-network/sbtc/issues/176 needs to be modified to reflect what this PR addresses I'll do that too.

Sure yeah this was only scoped for the block observer. I'm happy to leave the extensions to follow-ups. We can create those tickets when doing the skeleton implementation of the coordinator event loop, because that will create a natural context for them.

Reply:
Oh, well I added some:
* minting https://github.com/stacks-network/sbtc/issues/230
* burning https://github.com/stacks-network/sbtc/issues/231
* fee estimation https://github.com/stacks-network/sbtc/issues/233

We need more mint-many, deposit-reject, withdrawal-reject, and signing the transaction. Yeah let's add them later.

Reply:
Perfect, thanks for adding these :pray: